### PR TITLE
Added mso_schema_site_template_contract_service_graph resource and data-source

### PR DESCRIPTION
**Pull request by sajagana**
Note:

Added site_template_contract_service_graph resource and data-source to manage the Site template contract service graph

### CODE REVIEW

**Review by anvitha-jain — DISMISSED** [review]
LGTM

File: mso/resource_mso_site_template_contract_service_graph.go [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+	cont, err := msoClient.GetViaURL(fmt.Sprintf("api/v1/schemas/%s", serviceGraphRef["schemaId"].(string)))
+	if err != nil {
+		return err
+	}
+
+	schemaTemplateServiceGraphCont, _, err := getSchemaTemplateServiceGraph(cont, serviceGraphRef["templateName"].(string), serviceGraphName)
+	if err != nil {
+		return err
+	}
+	apiNodeRelationshipList, err := extractNodes(schemaTemplateServiceGraphCont)
```
Comment (sajagana) [inline mso/resource_mso_site_template_contract_service_graph.go:558]: Need to remove the variable err when the https://github.com/CiscoDevNet/terraform-provider-mso/pull/244 PR is merged with the master.

**Review by akinross — CHANGES_REQUESTED** [review]

File: examples/schema_site_template_contract_service_graph/main.tf [hunk 2]
```diff
@@ -66,7 +66,7 @@ resource "mso_schema_site" "schema_site_1" {
   template_name = data.mso_schema_template_service_graph.sg1.template_name
 }
 
-resource "mso_site_template_contract_service_graph" "example" {
+resource "mso_schema_site_template_contract_service_graph" "example" {
```
Comment (akinross) [inline examples/schema_site_template_contract_service_graph/main.tf:69]: I think you changed it incorrect, where it should be `mso_schema_site_contract_service_graph`

**Review by akinross — CHANGES_REQUESTED** [review]

File: examples/schema_site_template_contract_service_graph/main.tf (hunk 2 above)
Comment (akinross) [reply examples/schema_site_template_contract_service_graph/main.tf:69]: this applies to the entire name change

**Review by akinross — APPROVED** [review]
LGTM

**Review by samiib — DISMISSED** [review]
LGTM

File: examples/schema_site_contract_service_graph/main.tf [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  name          = "bd1"
+}
+
+data "mso_schema_template_bd" "bd2" {
+  schema_id     = data.mso_schema.schema2.id
+  template_name = "t2"
+  name          = "bd2"
+}
+
+data "mso_schema_template_contract" "c1" {
```
Comment (lhercot) [inline examples/schema_site_contract_service_graph/main.tf:51]: I think we need a full example with the template contract service graph part in as resources
I would also do the service graph part in it too (at both template and site levels) as resources

**Review by lhercot — CHANGES_REQUESTED** [review]

**Review by akinross — APPROVED** [review]
LGTM

**Review by shrsr — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**